integrationTests/testInitializer.go

```diff
@@ -301,16 +324,114 @@ func MakeDisplayTable(nodes []*TestProcessorNode) string {
+	str := fmt.Sprintf("%s Address: %s\n", tag, base64.StdEncoding.EncodeToString(accnt.AddressContainer().Bytes()))
 	str += fmt.Sprintf("  Nonce: %d\n", accnt.Nonce)
-	str += fmt.Sprintf("  Balance: %d\n", accnt.Balance)
+	str += fmt.Sprintf("  Balance: %d\n", accnt.Balance.Uint64())
 	str += fmt.Sprintf("  Code hash: %s\n", base64.StdEncoding.EncodeToString(accnt.CodeHash))
 	str += fmt.Sprintf("  Root hash: %s\n", base64.StdEncoding.EncodeToString(accnt.RootHash))
 
 	fmt.Println(str)
 }
 
+// CreateDummyAddress creates a random byte array with fixed size
+func CreateDummyAddress() state.AddressContainer {
+	buff := make([]byte, sha256.Sha256{}.Size())
+	_, _ = rand.Reader.Read(buff)
+
+	return state.NewAddress(buff)
+}
+
+// CreateDummyHexAddress returns a string encoded in hex with the given size
+func CreateDummyHexAddress(chars int) string {
+	if chars < 1 {
+		return ""
+	}
+
+	buff := make([]byte, chars/2)
+	_, _ = rand.Reader.Read(buff)
+
+	return hex.EncodeToString(buff)
+}
+
+// GenerateAddressJournalAccountAccountsDB returns an account, the accounts address, and the accounts database
+func GenerateAddressJournalAccountAccountsDB() (state.AddressContainer, state.AccountHandler, *state.AccountsDB) {
+	adr := CreateDummyAddress()
+	adb, _, _ := CreateAccountsDB(nil)
+	account, _ := state.NewAccount(adr, adb)
+
+	return adr, account, adb
+}
+
+// AdbEmulateBalanceTxSafeExecution emulates a tx execution by altering the accounts
+// balance and nonce, and printing any encountered error
+func AdbEmulateBalanceTxSafeExecution(acntSrc, acntDest *state.Account, accounts state.AccountsAdapter, value *big.Int) {
+
+	snapshot := accounts.JournalLen()
+	err := AdbEmulateBalanceTxExecution(acntSrc, acntDest, value)
+
+	if err != nil {
+		fmt.Printf("!!!! Error executing tx (value: %v), reverting...\n", value)
+		err = accounts.RevertToSnapshot(snapshot)
+
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+// AdbEmulateBalanceTxExecution emulates a tx execution by altering the accounts
+// balance and nonce, and printing any encountered error
+func AdbEmulateBalanceTxExecution(acntSrc, acntDest *state.Account, value *big.Int) error {
+
+	srcVal := acntSrc.Balance
+	destVal := acntDest.Balance
+
+	if srcVal.Cmp(value) < 0 {
+		return errors.New("not enough funds")
+	}
+
+	err := acntSrc.SetBalanceWithJournal(srcVal.Sub(srcVal, value))
+	if err != nil {
+		return err
+	}
+
+	err = acntDest.SetBalanceWithJournal(destVal.Add(destVal, value))
+	if err != nil {
+		return err
+	}
+
+	err = acntSrc.SetNonceWithJournal(acntSrc.Nonce + 1)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateTxProcessor returns a transaction processor
+func CreateTxProcessor(accnts state.AccountsAdapter) process.TransactionProcessor {
+	shardCoordinator := mock.NewMultiShardsCoordinatorMock(1)
+	txProcessor, _ := transaction.NewTxProcessor(accnts, TestHasher, TestAddressConverter, TestMarshalizer, shardCoordinator, &mock.SCProcessorMock{})
```

Comment:
suggestion to rename: createSimpleTxProcessor

Reply:
Renamed.